### PR TITLE
[CUR-80] Confirm before discarding in-progress Add Item flow

### PR DESCRIPTION
**Pull request by Akkkkkkki**
## Problem

In the Add Item flow, the close `X` and the Escape key both call `onClose()` unconditionally. The reset effect on `[isOpen]` then wipes `imagePreview`, `formData`, and `batchItems` on the next open.

A user who taps `X` by accident — easy on mobile, especially during `verify` or `batch-verify` — loses the entire in-flight item: the photo, the AI-analyzed metadata (a paid Gemini call), and any Story they had typed. There is no Undo and no "Are you sure?" guard.

This directly undermines two MVP constraints from `CLAUDE.md`:

- **Recoverable AI** — AI analysis must never become a hard blocker, and a mistap shouldn't burn a Gemini call.
- **Explicit outcomes** — users should never lose work silently.

Code references:
- Esc handler: `src/components/AddItemModal.tsx:283-286`
- Close `X`: `src/components/AddItemModal.tsx:1301-1307`
- Form reset on open: `src/components/AddItemModal.tsx:217-255`

## Approach

Routed both close affordances through a single `requestClose` guard scoped exactly to the AC: `step === 'verify' || 'batch-verify'` AND any of `imagePreview`, trimmed title, trimmed story, or non-empty `batchItems`.

- Added a `confirmingDiscard` state + `confirmRef`. When the guard fires, the body content swaps to a small in-modal confirmation panel ("Discard this item?" + `Keep editing` / `Discard`). The footer Save button is gated off so the only actions are the two confirm choices.
- Mirrored `hasInProgressWork` and `confirmingDiscard` into refs so the keydown listener (registered once per modal-open) reads current state without re-registering on every keystroke.
- Scoped the focus-trap root to `confirmRef` while confirming so Tab can't escape back to the form behind the prompt; `autoFocus` on `Keep editing` lands focus on the safer choice.
- Esc behavior: dirty → opens the confirmation; while confirming → dismisses the confirmation (not the modal); clean → closes the modal as before.
- Form state lives in React state (`formData`, `imagePreview`, `batchItems`), so unmounting the body and remounting on `Keep editing` preserves photo, AI metadata, title, and Story automatically.
- 4 new i18n keys added in both `en` and `zh` blocks (`discardItemTitle`, `discardItemDesc`, `discardItemAction`, `keepEditing`).
- 3 new Vitest regression tests in `tests/components/AddItemModal.test.tsx` covering: clean-step closes immediately, dirty `verify` shows confirm then `Keep editing` preserves both title + Story then `Discard` closes, and Esc on `batch-verify` toggles the confirmation without losing the batch.

## Why this approach

Smallest fix that satisfies every acceptance criterion without introducing a new component or new global state. The dirty check matches the AC exactly. Reusing the existing `panelSurfaceClasses` / `mutedText` / `borderClass` and the existing `Button` primary/ghost variants keeps the panel theme-aware across Gallery/Vault/Atelier without new tokens. Scoping the focus trap to a separate ref avoids needing `inert` polyfills or aria-hidden gymnastics, and re-mounting the form on Keep editing leans on React component state — the safest way to preserve work without a manual snapshot.

The recommended "swipe-to-dismiss" AC point is moot here: the modal isn't a swipeable bottom sheet (it's a full-height dialog with `rounded-t-3xl` on mobile but no gesture binding), so X + Esc are the only close affordances and both go through `requestClose`.

## Risks

Low–medium (Yellow tier). Touches the core Add Item flow but only the close path — Save, AI analyze, batch save, retry, manual-entry, image edit, and story prompt code paths are byte-identical. Form-state unmount/remount on `Keep editing` is safe because controlled-input values live in React state, not in the DOM. Verified in unit tests that the typed title + story survive a `Keep editing` round-trip.

Edge cases checked:
- Clean verify (no photo, no title, no story) → X / Esc close immediately, no confirmation
- Confirm UI doesn't show on `select-type`, `upload`, or `analyzing` steps (AC scope)
- Scroll-fade only renders when not confirming (test still passes)
- Existing CUR-44, CUR-45, and AddItemModal save-error tests unchanged and still pass

## How to verify

Vercel Preview: _populates when CI builds_

Steps:
1. Open the app, sign in, open `Add Item` for an existing collection.
2. Pick `Skip and add manually` → land on verify with empty form. Tap X → modal closes immediately (no prompt).
3. Reopen `Add Item`, upload a photo (AI analyzes), type some Story.
4. Tap X (or press Esc on desktop) → "Discard this item?" appears with `Keep editing` + `Discard`.
5. Tap `Keep editing` → returns to the form with photo, title, AI fields, and Story intact.
6. Tap X again → confirmation appears. Tap `Discard` → modal closes.
7. Switch to **中文**, repeat steps 2–6 → confirm UI reads `舍弃这件藏品？` with `继续编辑` / `舍弃`.
8. Repeat on `Rapid-Fire Mode` (batch): pick multiple photos, wait for analysis, press Esc → confirmation appears; press Esc again → confirmation dismisses and the batch list is intact.

Checks:
- [x] `npm run format:check` — passes
- [x] `npm test` — 575 passed, 2 skipped, 1 todo (3 new tests added)
- [x] `npm run build` — passes (vite 6.4.1, 1813 modules)
- [ ] `npm run test:e2e` — **could not run in this environment**: the bundled Playwright Chromium binary at `/opt/pw-browsers/chromium_headless_shell-1200/...` is missing and `npx playwright install chromium` is blocked by the sandbox network allowlist (`Download failure` from `playwright.download.prss.microsoft.com`). Same pre-existing environment limitation called out in PRs #222, #223, #228, #231, #233, #236, #243, #246, #248. CI's `Format / Unit / Build / E2E` job will run e2e against this commit.

## Screenshot / GIF

Not captured in-sandbox (no headless browser available — see e2e note). Visual change is a single in-modal confirmation panel that reuses existing theme tokens (`panelSurfaceClasses`, `mutedText`, `Button` primary/ghost) so Gallery/Vault/Atelier rendering should match the rest of the Add Item modal. Verifiable on the Vercel preview via the steps above.

## Linear

Closes CUR-80

## Rollback plan

Single commit; revert with:

```
git revert eaa804b
```

No schema, RLS, storage, feature flag, env var, or data changes.

---
_Generated by [Claude Code](https://claude.ai/code/session_01A7PxhSoBJGbQPResuS6NsM)_

### CODE REVIEW

File: src/components/AddItemModal.tsx [hunk 1]
```diff
@@ -344,6 +359,36 @@ export const AddItemModal: React.FC<AddItemModalProps> = ({
...
 
+  // Verify and batch-verify are the only steps where the user has already
+  // invested effort worth confirming before discard: a photo upload, AI-
+  // analyzed metadata, typed Story, or batch items lined up for save.
+  const hasInProgressWork =
+    (step === 'verify' || step === 'batch-verify') &&
+    (!!imagePreview ||
+      formData.title.trim().length > 0 ||
+      (formData.notes || '').trim().length > 0 ||
+      batchItems.length > 0);
```
[the remark this answers is not in the document]
Comment (Akkkkkkki) [reply src/components/AddItemModal.tsx:381]: Good catch — fixed in 8f7a908. `hasInProgressWork` now also flags `formData.rating > 0` and any non-internal entry in `formData.data` (skipping `_aiDescription`-style internal keys). Added two regression tests: rating-only and custom-field-only manual entry both trigger the discard prompt now. 577/577 tests pass.

---
_Generated by [Claude Code](https://claude.ai/code)_
